msat receive_payment and lnurl_withdraw

JssDWt opened the pull request:
Convert the interfaces for `receive_payment` and `lnurl_withdraw` to use msat parameters.
Note this is BREAKING CHANGE.

lnurl_withdraw now also takes a request struct.

The interface for the cli now also takes msat in receive_payment, that may be confusing for regular cli users.

Uncertainties:
- I'm unsure about the flutter code changes. I ran the flutter_rust_bridge and updated breez_sdk.dart manually.

ok300 commented on libs/sdk-core/src/breez_services.rs:
```diff
@@ -416,8 +411,7 @@ impl BreezServices {
+    /// * `req_data` - Request parameters for receiving a payment
     pub async fn receive_payment(
         &self,
         req_data: ReceivePaymentRequest,
```
This can become `req` just like in `lnurl_withdraw(`

JssDWt replied:
done

ok300 commented on libs/sdk-core/src/breez_services.rs:
```diff
@@ -2153,7 +2145,7 @@ pub(crate) mod tests {
         let ln_invoice = receiver
             .receive_payment(ReceivePaymentRequest {
-                amount_sats: 3000,
+                amount_msat: 3000000,
```
This can be written as `3_000_000` to improve readability

JssDWt replied:
done

ok300 commented on tools/sdk-cli/src/command_handlers.rs:
```diff
@@ -336,29 +336,33 @@ pub(crate) async fn handle_command(
-                    let user_input_max_sat = wd.max_withdrawable / 1000;
-                    let user_input_min_sat = 2001;
+                    let user_input_max_msat = wd.max_withdrawable;
+                    let user_input_min_msat = 2001000;
```
`2_001_000`

JssDWt replied:
done

ok300 reviewed (commented):
Just a few NITs, otherwise looks good!

dangeross reviewed (changes requested):
~I think you still need to update the sdk-cli command handler for LnurlWithdraw~ Sorry, misread

JssDWt commented:
> @JssDWt We use 110 as line length for c-breez & sdk-flutter. That's why there are a lot of indentation changes. Please run `dart format --fix -l 110 .` from `sdk-flutter` directory.
> 
> Other than that, they look good(_except the reqData->req naming Carlos mentioned_).

This is now done

erdemyerebasmaz commented on libs/sdk-flutter/lib/breez_sdk.dart:
```diff
@@ -273,19 +272,11 @@ class BreezSDK {
   /// Second step of LNURL-withdraw. The first step is `parse()`, which also validates the LNURL destination
   /// and generates the `LnUrlWithdrawRequestData` payload needed here.
   ///
-  /// This call will validate the given `amount_sats` against the parameters
+  /// This call will validate the given `amount_msat` against the parameters
   /// of the LNURL endpoint (`req_data`). If they match the endpoint requirements, the LNURL withdraw
   /// request is made. A successful result here means the endpoint started the payment.
```
`req_data` changes will be handled on #497 as discussed on #536

That PR also has a great example of this change:

> Documentation has to reference req(LnUrlWithdrawRequest).

dangeross replied:
@JssDWt can you update `sdk-react-native/example/App.js` also?

```
                const receivePaymentResult = await receivePayment({
                    amountMsat: 100000000,
                    description: "Hello world",
                    expiry: 3600,
                    cltv: 144,
                    useDescriptionHash: true,
                    openingFeeParams: openChannelFeeResult.usedFeeParams
                })
```

roeierez reviewed (approved):
LGTM

ok300 commented:
Looks like `cargo fmt` is the only thing missing.

JssDWt commented:
Renamed request parameter to `request`
Renamed `req_data` to `data`
Added documentation comments on LnurlWithdrawRequest struct fields
Modified react native example to match the new field name.

JssDWt commented:
@erdemyerebasmaz Can you do another pass given the changes mentioned here https://github.com/breez/breez-sdk/pull/535#issuecomment-1771526550

I lost track a bit on which naming should be used where.

erdemyerebasmaz commented:
> @erdemyerebasmaz Can you do another pass given the changes mentioned here [#535 (comment)](https://github.com/breez/breez-sdk/pull/535#issuecomment-1771526550)
> 
> I lost track a bit on which naming should be used where.

`request` parameter should stay as `req`, we decided against this change on https://github.com/breez/breez-sdk/pull/497#discussion_r1343410803

Other changes look good. If you prefer not to go into this again we can merge and I can address this as part of #497.

JssDWt commented:
@erdemyerebasmaz I merged it, let's put the naming in https://github.com/breez/breez-sdk/pull/497
